tools/runners/run-app.js

```diff
@@ -172,18 +169,16 @@ Object.assign(AppProcess.prototype, {
     }
     if (self.settings) {
       env.METEOR_SETTINGS = self.settings;
-    } else {
+    } else if (env.METEOR_SETTINGS && env.NODE_ENV === 'development') {
```

Comment:
The new if statement ensures that `METEOR_SETTINGS` is deleted only in development. In other scenarios it will be applied.

Reply:
With the react refresh integration, we've found that NODE_ENV isn't always set to the correct value matching the build mode. If the build mode is available here, it would be more reliable.